fem/tmop.hpp

```diff
@@ -677,6 +677,44 @@ class TargetConstructor
                                       const IntegrationRule &ir,
                                       const Vector &elfun,
                                       DenseTensor &Jtr) const;
+
+   virtual void ComputeElementTargetsGradient(int e_id, const FiniteElement &fe,
+                                              const IntegrationRule &ir,
+                                              const Vector &elfun,
+                                              IsoparametricTransformation &Tpr,
+                                              DenseTensor &dJtr) const;
+};
+
+class TMOPMatrixCoefficient  : public MatrixCoefficient
+{
+public:
+   explicit TMOPMatrixCoefficient(int dim) : MatrixCoefficient(dim, dim) { }
+
+   TMOPMatrixCoefficient(int h, int w) : MatrixCoefficient(h, w)  { }
+
+   void SetTime(double t) { time = t; }
+   double GetTime() { return time; }
+
+   int GetHeight() const { return height; }
+   int GetWidth() const { return width; }
+   // For backward compatibility
+   int GetVDim() const { return width; }
```

Comment:
These function don't need to be here, we only need the first constructor and the Evals.